Research/ResearchUI/iOS/Views/RSDTemplateImageView.swift

```diff
@@ -0,0 +1,313 @@
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+/// The template image button is designed as a button that shows a template image within a
+/// button. There are different supported style-types for the button that can be used with the
+/// design system to set up the color of the button and the default tint for the image within.
+@IBDesignable
+open class RSDTemplateImageButton: UIButton, RSDViewDesignable {
+    
+    /// The corner radius for the image view. Default is -1 which will result in a circle.
+    @IBInspectable public var cornerRadius: CGFloat {
+        get { return _cornerRadius ?? -1 }
+        set {
+            _cornerRadius = (newValue >= 0) ? newValue : nil
+            updateCornerRadius()
+        }
+    }
+    fileprivate var _cornerRadius: CGFloat?
+    fileprivate var _circleSize: CGFloat!
+    
+    /// The width of the border for views with a border. Default = 0.0
+    /// The border width must be set to a value > 0 and the border color must be set to show a border.
+    @IBInspectable public var borderWidth: CGFloat = 0.0 {
+        didSet {
+            self.layer.borderWidth = borderWidth
+        }
+    }
+    
+    /// The color of the border for views with a border. Default = nil
+    /// The border width must be set to a value > 0 and the border color must be set to show a border.
+    @IBInspectable public var borderColor: UIColor? = nil {
+        didSet {
+            self.layer.borderColor = borderColor?.cgColor
+        }
+    }
+    
+    /// Should the view display a shadow? Default = false
+    @IBInspectable public var hasShadow: Bool = false {
+        didSet {
+            updateShadow()
+        }
+    }
+    
+    override open var isEnabled: Bool {
+        didSet {
+            // If the alpha component is used to set this as hidden, then don't do anything.
+            guard alpha > 0.1 else { return }
+            self.alpha = isEnabled ? CGFloat(1) : CGFloat(0.35)
+        }
+    }
+    
+    override open var isHighlighted: Bool {
+        didSet {
+            // If the alpha component is used to set this as hidden, then don't do anything.
+            guard alpha > 0.1 else { return }
+            
+            guard let colorTile = self.buttonBackground,
+                let background = self.backgroundColorTile
+                else {
+                    // show as highlighted by lowering opacity unless alpha is used to set hidden
+                    self.alpha = isHighlighted ? CGFloat(1) : CGFloat(0.8)
```

Comment:
this looks like it lowers the opacity if *not* highlighted (which seems correct to me)--is the above comment misleading? or am I misunderstanding something?

Reply:
Whoops, nope. that's a bug. Thanks!